static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -284,22 +280,55 @@ function CustomViewsIssueListHeaderTabsContent({
               ...queryParamsWithPageFilters,
               viewId: undefined,
             },
-          })
+          }),
+          {replace: true}
         );
         trackAnalytics('issue_views.shared_view_opened', {
           organization,
           query,
         });
-        return;
       }
       return;
     }
     if (query) {
       tabListState?.setSelectedKey(TEMPORARY_TAB_KEY);
+      navigate(
+        normalizeUrl({
+          ...location,
+          query: {
+            ...queryParamsWithPageFilters,
+            viewId: undefined,
+          },
+        }),
+        {replace: true}
+      );
       return;
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [navigate, organization.slug, query, sort, viewId, tabListState]);
+  }, []);
```

Comment:
I have refactored this useEffect to _only run on first render_. Previously, it was responsible for two things: 

1. Making sure that the query params align with the tab that's selected 
2. Making sure that the unsaved changes are being registered properly every time a new query is loaded

This led to a lot of double work being done: the tabs component already handles updating the url query properly when the user switches between views, so the navigates in this useEffect were unnecessary and redundant. Now, this useEffect is only responsible for making sure the query params align with the tab that's selected **when the component initially loads**, thus allowing the tabs component to handle navigation thereon after. 

The new useEffect below handles updating the unsaved changes when the search query changes.

Reply:
Removing dependencies from the array is often a source of problems, so you want to be careful here. I think this may have introduced a bug where if you click to a different view tab, then click "Issues" in the sidebar you redirect to `/issues`, but your tab does not change (and has the temp tab visual). You want to be careful of any values that might change after first render

Reply:
@malwilley good catch, I can reproduce this and it's 100% caused by the removal of these dependencies.

I'm curious if you think removing these dependencies is the right option at all. My thought process was that it was a way too overloaded - it was handling both setting the query params on mount, and on most tab actions like switching. I figured this led to some conflicting behavior with the tabListState that react-aria' provides, like double navigating to the same tab every time we switched (though that's still occurring in this branch). I can't tell if I'm being overly nervous with how much the scope of this useEffect has grown. 🤔

Reply:
Ideally we'd have the useEffect run on every dep change (and remove that eslint ignore line). To decrease the scope, you might be able to separate it into multiple different useEffect's. That is, if that's possible at all.

Reply:
I updated the useEffect to include all dependencies and ensured it does not infinitely loop, and I removed the second useEffect. If it does infinitely render, would this be caught in tests somehow?